addon/webextension/background/analytics.js

```diff
@@ -8,6 +8,33 @@ this.analytics = (function() {
   let telemetryPrefKnown = false;
   let telemetryPref;
 
+  function sendTiming(timingLabel, timingVar, timingValue) {
+    // sendTiming is only called in response to sendEvent, so no need to check
+    // the telemetry pref again here.
+    let timingCategory = "addon";
+    let url = main.getBackend() + "/timing";
+    let req = new XMLHttpRequest();
+    req.open("POST", url);
+    req.setRequestHeader("content-type", "application/json");
+    req.onload = catcher.watchFunction(() => {
+      if (req.status >= 300) {
+        let exc = new Error("Bad response from POST /timing");
+        exc.status = req.status;
+        exc.statusText = req.statusText;
+        console.error(exc);
+        // TODO: send the error to sentry, maybe?
```

Comment:
This should reject a promise. Also it should create a promise! Same as sendEvent. I'm not sure how consistently we watch the sendEvent promise, arguably we should have a wrapped version of these functions for the not-uncommon case where we want to send-and-forget (with `catcher.watchFunction(sendTiming, true)`)